web/react/components/post.jsx

```diff
@@ -223,6 +244,10 @@ export default class Post extends React.Component {
                                 posts={posts}
                                 handleCommentClick={this.handleCommentClick}
                                 retryPost={this.retryPost}
+                                setHasEmbedState={this.setHasEmbedState}
+                                toggleEmbedVisibility={this.toggleEmbedVisibility}
+                                hasEmbed={this.state.hasEmbed}
+                                embedVisible={this.state.embedVisible}
```

Comment:
It looks like the PostBody is the only component that makes use of the embed visibility. Do you think it's worth having the extra logic to keep the state stored in the Post itself rather just encapsulating it all within PostBody?